mail/spring-cloud-starter-stream-source-mail/src/main/java/org/springframework/cloud/stream/app/mail/source/MailSourceConfiguration.java

```diff
@@ -0,0 +1,107 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.mail.source;
+
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.trigger.TriggerConfiguration;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.mail.Mail;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.mail.transformer.MailToStringTransformer;
+import org.springframework.integration.scheduling.PollerMetadata;
+
+/**
+ * A source module that listens for mail and emits the content as a message payload.
+ *
+ * @author Amol
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties({ MailSourceProperties.class })
+@Import({ TriggerConfiguration.class })
+public class MailSourceConfiguration {
+
+	@Autowired
+	@Qualifier("defaultPoller")
+	PollerMetadata defaultPoller;
+
+	@Autowired
+	@Bindings(MailSourceConfiguration.class)
+	Source source;
+
+	@Autowired
+	MailSourceProperties properties;
+
+	@Bean
+	public IntegrationFlow mailInboundFlow() {
+		Properties javaMailProperties = new Properties();
+		String mailURL = properties.getProtocol() + "://" + properties.getUsername() + ":"
+				+ properties.getPassword() + "@" + properties.getHost() + ":"
+				+ properties.getPort() + "/" + properties.getFolder();
+
+		javaMailProperties.putAll(properties.parsePropertiesString());
+
+		IntegrationFlowBuilder flowBuilder;
+		if (properties.getProtocol().equalsIgnoreCase("imaps")
+				|| properties.getProtocol().equalsIgnoreCase("imap")) {
+			flowBuilder = IntegrationFlows.from(
+					Mail.imapInboundAdapter(mailURL)
+							.javaMailProperties(javaMailProperties)
+							.shouldMarkMessagesAsRead(properties.isMarkAsRead())
+							.shouldDeleteMessages(properties.isDelete()),
+					new Consumer<SourcePollingChannelAdapterSpec>() {
+
+						@Override
+						public void accept(
+								SourcePollingChannelAdapterSpec sourcePollingChannelAdapterSpec) {
+							sourcePollingChannelAdapterSpec.poller(defaultPoller);
+						}
+					});
+
+		}
+		else {
+			flowBuilder = IntegrationFlows.from(
+					Mail.pop3InboundAdapter(mailURL)
```

Comment:
The `if...else` code could be simplified, when we extract `Mail...Adapter()` variable.
But yeah, minor.